docs/database-setup.md

```diff
@@ -1,51 +1,80 @@
-
-## Creating a database
-
     $ sudo -u postgres createdb -O <username> <database name>
-
-This database will be owned by the user you've specified. We recommend the
-database name be the same as the username.
-
-## Testing access
-
-To test that you're able to connect (type `\q` to quit after you log in):
-
+    # test it's there
     $ psql -U <username> -h localhost <database name>
 
-You can omit the database name here if it's the same as the username.
-
-## Database URL
-
-Finally, once everything is figured out, the `DATABASE_URL` parameter you're
-prompted for during install will look something like this:
-
-    pgsql://<username>:<password>@localhost/<database name>?serverVersion=9.5
-
-The server version should be set to the version of PostgreSQL provided by your
-distro. For Ubuntu 16.04, it's '9.5'. If unsure, you can omit this parameter
-entirely and things should still work.
-
-## Other resources
-
-* <https://help.ubuntu.com/community/PostgreSQL>
-* <https://www.postgresql.org/docs/9.5/static/app-createdb.html>
-* <https://www.postgresql.org/docs/9.5/static/app-createuser.html>
+Modify your .env with following URL:
+
+    pgsql://<username>:<password>@localhost/<database name>?serverVersion=9.6
+
+Enjoy!
+
+## Real Deal
+
+Still here? Awesome. Get Some!
+
+     // Create Users
+     CREATE ROLE youwish_devuser WITH LOGIN PASSWORD 'oooh_heees_trying';
+     CREATE ROLE youwish_testuser WITH LOGIN PASSWORD 'oooh_heees_trying';
+     CREATE ROLE youwish_produser WITH LOGIN PASSWORD 'oooh_heees_trying';
+     CREATE ROLE youwish_prodadmin WITH LOGIN PASSWORD 'oooh_heees_trying';
+     // Give all other user's permissions to DB master/root user
+     GRANT youwish_devuser TO youwish_root;
+     GRANT youwish_testuser TO youwish_root;
+     GRANT youwish_produser TO youwish_root;
+     GRANT youwish_prodadmin TO youwish_root;
+     // Give admin prod role everything prod-readWrite has
+     GRANT youwish_produser TO youwish_prodadmin;
+     // Create Databases
+     CREATE DATABASE youwish_prod;
+     REVOKE ALL ON DATABASE youwish_prod FROM public;
+     CREATE DATABASE youwish_dev;
+     REVOKE ALL ON DATABASE youwish_dev FROM public;
+     CREATE DATABASE youwish_test;
+     REVOKE ALL ON DATABASE youwish_test FROM public;
+     // Prod and ProdAdmin can both connect to Prod, but Owner is ProdAdmin
+     ALTER DATABASE youwish_prod OWNER to youwish_prodadmin;
+     GRANT CONNECT ON DATABASE youwish_prod TO youwish_prodadmin;
+     GRANT CONNECT ON DATABASE youwish_prod TO youwish_produser;  
+     // Dev and Test user===admin
+     ALTER DATABASE youwish_dev OWNER to youwish_devuser;
+     ALTER DATABASE youwish_test OWNER to youwish_testuser;
+     GRANT CONNECT ON DATABASE youwish_dev TO youwish_devuser;
+     GRANT CONNECT ON DATABASE youwish_test TO youwish_testuser;
+     // For each db - create schema and default priveleges
+     \connect youwish_prod
+     CREATE SCHEMA schma AUTHORIZATION youwish_prodadmin;
```

Comment:
We could separate everything into different schema within the same database... but this works just fine as well.  MySQL is the oddball in DB land in that it only has 3 layers of FQDN - everyone else has 4.  host.database.schema.object  MySQL combines database and schema into 1.  So for this, we can create a different database for everyone and then use the same schema name (could even be public actually).  Or we can create a different schema and they all live within the same DB.  Or it could be specific db and specific schema.  All of them work, but having both be unique seems like an extra parameter setting that wouldn't be needed.  It also won't really hurt anything from a DBA standpoint.  It's just a little bit of extra typing.

Looking at the backup/restore part - we want a different DB for everyone.  That will make things the easiest.  Let's specify the schema that we want to use here (unless schma was the actual name you intended to use).  I would probably name it something like "gundeal" or something that works in all databases to know that it is where things would be expected.

Reply:
yeah, schma was intended. Don't think it needs to be secret